hyper/http11/connection.py

```diff
@@ -75,6 +79,21 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self.ssl_context = ssl_context
         self._sock = None
 
+        # Setup proxy details if applicable.
+        if proxy_host:
+            if proxy_port is None:
+                try:
+                    self.proxy_host, self.proxy_port = proxy_host.split(':')
```

Comment:
Agreed this won't work with IPv6 addresses; however, the rest of the codebase or at least the http11 and http20 constructors have similar logic when just parsing the host and port.  Should these all be updated as part of this effort and test cases added as well?